placement/placement.go

```diff
@@ -659,16 +629,15 @@ func markShardAvailable(p Placement, instanceID string, shardID uint32) (Placeme
 
 	sourceShards.Remove(shardID)
 	if sourceShards.NumShards() == 0 {
-		return ClonePlacement(p).
-			SetInstances(RemoveInstanceFromList(p.Instances(), sourceInstance.ID())), nil
+		return p.SetInstances(RemoveInstanceFromList(p.Instances(), sourceInstance.ID())), nil
```

Comment:
No need to clone any more?

Reply:
nope, the 2 functions that are calling this function did a clone already